pkg/mcs/tso/server/server.go

```diff
@@ -19,37 +19,86 @@ import (
 	"flag"
 	"net/http"
 	"os"
+	"sync"
+	"time"
 
 	grpcprometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
 	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/diagnosticspb"
+	"github.com/pingcap/kvproto/pkg/tsopb"
 	"github.com/pingcap/log"
 	bs "github.com/tikv/pd/pkg/basicserver"
 	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/pkg/member"
 	"github.com/tikv/pd/pkg/tso"
+	"github.com/tikv/pd/pkg/utils/grpcutil"
 	"github.com/tikv/pd/pkg/utils/logutil"
 	"github.com/tikv/pd/pkg/utils/metricutil"
+	"github.com/tikv/pd/pkg/utils/tsoutil"
 	"go.etcd.io/etcd/clientv3"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-// Server is the TSO server, and it implements bs.Server.
-// nolint
+// If server doesn't implement all methods of bs.Server, this line will result in a clear
+// error message like "*Server does not implement bs.Server (missing Method method)"
+var _ bs.Server = (*Server)(nil)
+
+// Server is the TSO server, and it implements bs.Server and tso.GrpcServer.
 type Server struct {
-	ctx context.Context
+	diagnosticspb.DiagnosticsServer
+
+	// Server start timestamp
+	startTimestamp int64
+
+	ctx       context.Context
+	name      string
+	clusterID uint64
+	// etcd client
+	client *clientv3.Client
+	// http client
+	httpClient          *http.Client
+	member              *member.Member
+	tsoAllocatorManager *tso.AllocatorManager
+	// Store as map[string]*grpc.ClientConn
+	clientConns sync.Map
+	// Store as map[string]chan *tsoRequest
+	tsoDispatcher sync.Map
+	// Callback functions for different stages
+	// startCallbacks will be called after the server is started.
+	startCallbacks []func()
+	// leaderCallbacks will be called after the server becomes leader.
+	leaderCallbacks []func(context.Context)
```

Comment:
Will they be used in the later implementation?

Reply:
Yes, will be in the 3rd pr split from #5949